### PR TITLE
[SPARK-36490][SQL] Make from_csv/to_csv to handle timestamp_ntz type properly

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes an issue that `from_csv` and `to_csv` cannot handle `timestamp_ntz` type properly.
In the current master, to_csv/from_csv can handle timestamp type like as follows.
```
SELECT to_csv(struct(TIMESTAMP"2021-11-23 11:22:33")); 
2021-11-23T11:22:33.000+09:00
```
```
SELECT from_csv("2021-11-23 11:22:33", "a TIMESTAMP"); 
{"a":2021-11-23 11:22:33}
```

But they cannot handle timestamp_ntz type properly.
```
SELECT to_csv(struct(TIMESTAMP_NTZ"2021-11-23 11:22:33"));
-- 2021-11-23T11:22:33.000 is expected.
1637666553000000
```
```
SELECT from_csv("2021-11-23 11:22:33", "a TIMESTAMP_NTZ");
21/08/12 16:12:49 ERROR SparkSQLDriver: Failed in [SELECT from_csv("2021-11-23 11:22:33", "a TIMESTAMP_NTZ")]
java.lang.Exception: Unsupported type: timestamp_ntz
        at org.apache.spark.sql.errors.QueryExecutionErrors$.unsupportedTypeError(QueryExecutionErrors.scala:777)
        at org.apache.spark.sql.catalyst.csv.UnivocityParser.makeConverter(UnivocityParser.scala:234)
        at org.apache.spark.sql.catalyst.csv.UnivocityParser.$anonfun$valueConverters$1(UnivocityParser.scala:134)
```

### Why are the changes needed?

Bug fix.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out [hunk 1]
```diff
@@ -722,7 +722,6 @@ Unsupported type: timestamp_ntz
 -- !query
 select from_csv('26/October/2015', 't Timestamp', map('timestampFormat', 'dd/MMMMM/yyyy'))
 -- !query schema
-struct<>
+struct<from_csv(26/October/2015):struct<t:timestamp_ntz>>
 -- !query output
-java.lang.Exception
-Unsupported type: timestamp_ntz
+{"t":null}
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out:727]: Why null? Which issue does the input have? No time part?

File: sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp.sql.out [hunk 2]
```diff
@@ -705,7 +705,6 @@ Unsupported type: timestamp_ntz
 -- !query
 select from_csv('26/October/2015', 't Timestamp', map('timestampFormat', 'dd/MMMMM/yyyy'))
 -- !query schema
-struct<>
+struct<from_csv(26/October/2015):struct<t:timestamp_ntz>>
 -- !query output
-java.lang.Exception
-Unsupported type: timestamp_ntz
+{"t":null}
```
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp.sql.out:710]: The same question here, why null?

**Review by MaxGekk — COMMENTED** [review]
LGTM in general except of a minor comment.

File: sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out (hunk 1 above)
Comment (MaxGekk) [reply sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out:727]: ok. I have figured out: https://github.com/apache/spark/pull/28684/files#diff-d95bef2a8044eee384b21ba62b37d726c83eb7f48e768279db898197af3f5dd4R233

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @sarutak .

File: sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out (hunk 1 above)
Comment (sarutak) [reply sql/core/src/test/resources/sql-tests/results/timestampNTZ/timestamp-ansi.sql.out:727]: Sorry, I should note about null here. Yes, `MMMMM` is an invalid format and I think it's intentional.

**Pull request closed without merging**